### ISSUE TITLE
effecient bob emitter

### ISSUE THREAD

**Issue by aeyakovenko** [solana-labs/solana#950]
in PR #739, packets, or incoming batches of transactions lock memory for their account keys.  This allows for an optimization during blob generation.
1. packet (batch of txs) locks memory for each key in each tx
2. tx's are processed
3. valid tx's are written to blob for output to ledger
4. when blob is ready (full or timeout has passed), and no more txs will be written into this blob, the memory for each key in the blob is unlocked.
5. blob written to network/ledger

This pipeline only has a single copy per TX and packs transactions into parallelizable blobs as efficiently as possible within the timeout window from step 4.

**Comment by aeyakovenko** [comment]
@rob-solana what the code does right now is construct entries (allocation + copy), then stick them into blobs (a copy).  instead of writing into blobs then sending them out to IO.  since you are not binning at the IO boundary, you have no idea how to construct the entry in such a way that it fits in whatever space is left in the blob.

**Comment by rob-solana** [comment]
this issue and the mt-bank PR don't look in sync anymore.  are they?

**Comment by garious** [comment]
No, one of @aeyakovenko's comments on that PR said that he considered it out of scope.

**Comment by mvines** [comment]
@aeyakovenko, are we optimized?  Anything more that needs to be done here?
